libs/datamodel/core/src/common/constraint_names.rs

```diff
@@ -127,7 +127,12 @@ impl ConstraintNames {
         let column_names: Vec<&str> = ri
             .fields
             .iter()
-            .map(|field_name| model.find_scalar_field(field_name).unwrap().final_database_name())
+            .map(|field_name| {
+                model
+                    .find_scalar_field(field_name)
+                    .map(|field| field.final_database_name())
+                    .unwrap_or(field_name)
```

Comment:
I think we should keep the unwrap there, if we try to find a field that doesn't exist, it's a bug.

Reply:
If we want to write a data model with relations that don't point to any specific scalar field anymore, we cannot panic if the scalar field doesn't exist.

Reply:
I still uncomfortable merging this. We should at least have a comment explaining why we do that.

Reply:
Just write a comment if we keep it, and let's merge.